src/Symfony/Bundle/SecurityBundle/DataCollector/SecurityDataCollector.php

```diff
@@ -136,15 +136,33 @@ public function collect(Request $request, Response $response, \Exception $except
 
         // collect voters and access decision manager information
         if ($this->accessDecisionManager instanceof TraceableAccessDecisionManager) {
-            $this->data['access_decision_log'] = $this->accessDecisionManager->getDecisionLog();
-            $this->data['voter_strategy'] = $this->accessDecisionManager->getStrategy();
-
-            foreach ($this->accessDecisionManager->getVoters() as $voter) {
-                $this->data['voters'][] = $this->hasVarDumper ? new ClassStub(\get_class($voter)) : \get_class($voter);
+            $decisionLog = $this->accessDecisionManager->getDecisionLog();
+
+            $voterDetails = array();
+            $voters = array();
+            foreach ($decisionLog as $key => $log) {
+                $voterDetails[$key] = array();
+                foreach ($log['voterDetails'] as $voterClass => $voterVote) {
+                    $classData = $this->hasVarDumper ? new ClassStub($voterClass) : $voterClass;
+                    $voterDetails[$key][] = array(
+                        'class' => $classData,
+                        'vote' => $voterVote,
+                    );
+
+                    if (!\in_array($classData, $voters)) {
```

Comment:
I think we should rather save this function call at all and iterate `$this->accessDecisionManager->getVoters()` instead to populate `$this->data['voters']` (just like it was done before)

Reply:
Modification done :)

Reply:
@xabbuh I set this discussion as unresolved since `$this->accessDecisionManager->getVoters()` returns `TraceableVoter` instances, so `TraceableVoter` is displayed in the profiler instead of `ExpressionVoter` for instance. It was the reason I did that in this way. I'm going to fix it :)

Reply:
Fixed and tests added.